### PR TITLE
[Benchmark] Add `--async-engine` option to benchmark_throughput.py

### PR DESCRIPTION
**Pull request by njhill**
Uses AsyncLLMEngine interface (rather than LLM)

Will use decoupled front-end depending on whether or not `--disable-frontend-multiprocessing` is also specified.

### CODE REVIEW

File: vllm/entrypoints/openai/api_server.py [hunk 1]
```diff
@@ -192,6 +207,8 @@ async def build_async_engine_client(
             from prometheus_client import multiprocess
             multiprocess.mark_process_dead(rpc_server_process.pid)
 
+            async_engine_client = None  #TODO
```
Comment (robertgshaw2-neuralmagic) [inline vllm/entrypoints/openai/api_server.py:210]: why TODO?

File: vllm/entrypoints/openai/api_server.py [hunk 2]
```diff
@@ -104,22 +120,21 @@ async def build_async_engine_client(
...
+    if (model_is_embedding(engine_args.model, engine_args.trust_remote_code,
+                           engine_args.quantization)
+            or disable_frontend_multiprocessing):
         async_engine_client = AsyncLLMEngine.from_engine_args(
             engine_args, usage_context=UsageContext.OPENAI_API_SERVER)
-        yield async_engine_client
+        try:
+            yield async_engine_client
+        finally:
+            async_engine_client.shutdown_background_loop()
```
Comment (robertgshaw2-neuralmagic) [inline vllm/entrypoints/openai/api_server.py:136]: why is this needed? (curious)

File: vllm/entrypoints/openai/api_server.py (hunk 1 above)
Comment (njhill) [reply vllm/entrypoints/openai/api_server.py]: Removed this now

File: vllm/entrypoints/openai/api_server.py (hunk 2 above)
Comment (njhill) [reply vllm/entrypoints/openai/api_server.py]: To ensure clean shutdown in the non-decoupled case

File: benchmarks/benchmark_throughput.py [hunk 3]
```diff
@@ -240,7 +330,14 @@ def main(args: argparse.Namespace):
...
             args.use_v2_block_manager, args.download_dir, args.load_format,
-            args.disable_async_output_proc)
+            args.disable_async_output_proc
+        ]
+
+        if args.async_engine:
+            run_args.append(args.disable_frontend_multiprocessing)
+            elapsed_time = uvloop.run(run_vllm_async(*run_args))
+        else:
+            elapsed_time = run_vllm(*run_args)
```
Comment (comaniac) [inline benchmarks/benchmark_throughput.py:340]: Nit: better to use kwargs

Comment (comaniac) [inline benchmarks/benchmark_throughput.py:340]: Nit: better to use kwargs

**Review by comaniac — APPROVED** [review]
LGTM. Just a nit

File: benchmarks/benchmark_throughput.py (hunk 3 above)
Comment (njhill) [reply benchmarks/benchmark_throughput.py:340]: Agree, but the list was already passed as regular args so this involved minimal changes.
